04_TS_decorator/classes/building.ts

```diff
@@ -0,0 +1,57 @@
+import { Student } from "../types/types";
+
+export class Building {
+    constructor (protected foundationYear: number, protected city: string) { }
+
+    public getBuildingInfo() {
+        console.log(`This building was found in ${this.foundationYear} in ${this.city}.`);
+    }
+}
+
+export class Univercity extends Building {
+    constructor (foundationYear: number, city: string, protected name: string) {
+        super(foundationYear, city);
+    }
+
+    protected students: String[] = [];
+    protected studentsAge: number[] = []
+
+    public getStudentsList() {
+        if(this.students.length) {
+            this.students.forEach(item => {console.log(item);})    
+        } else {
+            console.log(`The studying year hasn't begun yet in ${this.name} university.`);    
+        }
+    }
+
+    public getYoungestStudentAge() {
+        if(this.students.length) {
+            console.log(`The youngest student we have in ${this.name} university is ${Math.min(...this.studentsAge)} years old.`);
+        } else {
+            console.log(`We are still waiting for new students.`);    
+        }
+    }
+
+    public assignStudents(... args: Student[]) { 
+        args.forEach((item) => {
+            let fullName: String = `${item.name} ${item.surname}`
+            if(!this.students.includes(fullName) && item.age >= 16) {
+                this.students.push(fullName)
+                this.studentsAge.push(item.age)
```

Comment:
So according to your logic if student name is already in the `students` list and his/her age is `>=16` it shouldn't  be added to the university? Please reread the task. Duplicates are allowed in the `students` list, but prohibited in the `studentsAge` list. The only reason why student shouldn't be added to the university is his age.